contracts/utils/Strings.sol

```diff
@@ -115,4 +121,89 @@ library Strings {
     function equal(string memory a, string memory b) internal pure returns (bool) {
         return bytes(a).length == bytes(b).length && keccak256(bytes(a)) == keccak256(bytes(b));
     }
+
+    /**
+     * @dev Parse a decimal string and returns the value as a `uint256`.
+     *
+     * This function will revert if:
+     * - the string contains any character that is not in [0-9].
+     * - the result does not fit in a `uint256`.
+     */
+    function toUint(string memory input) internal pure returns (uint256) {
+        bytes memory buffer = bytes(input);
+
+        uint256 result = 0;
+        uint256 bufferLength = buffer.length;
+        for (uint256 i = 0; i < bufferLength; ++i) {
+            result *= 10; // will revert if overflow
+            result += _parseChr(buffer[i], 10);
+        }
+        return result;
+    }
+
+    /**
+     * @dev Parse a decimal string and returns the value as a `int256`.
+     *
+     * This function will revert if:
+     * - the string contains any character (outside the prefix) that is not in [0-9].
+     * - the result does not fit in a `int256`.
+     */
+    function toInt(string memory input) internal pure returns (int256) {
+        bytes memory buffer = bytes(input);
+
+        // check presence of a negative sign.
+        bool isNegative = bytes1(buffer) == 0x2d;
```

Comment:
what about the true negative char "−" (0x2212)?

Reply:
is that comonly used/supported? 
- In my local node instance, using it causes a `Uncaught SyntaxError: Invalid or unexpected token`
- In my local python instance, using it causes a `SyntaxError: invalid character '−' (U+2212)`

I honestly don't think we should bother with non ascii characters.